ingestr/src/appsflyer/_init_.py

```diff
@@ -11,14 +13,38 @@
 def appsflyer_source(
     api_key: str, start_date: str, end_date: str
 ) -> Iterable[DltResource]:
+    start_date_obj = ensure_pendulum_datetime(start_date)
     client = AppsflyerClient(api_key)
 
-    @dlt.resource(write_disposition="merge", merge_key="install_time")
-    def campaigns() -> Iterable[TDataItem]:
-        yield from client.fetch_campaigns(start_date, end_date)
+    @dlt.resource(write_disposition="merge", merge_key="Install Time")
+    def campaigns(
+        updated=dlt.sources.incremental('["Install Time"]', start_date_obj.isoformat()),
+    ) -> Iterable[TDataItem]:
+        current_start_time = datetime.fromisoformat(updated.start_value).date()
+        end_date_time = datetime.fromisoformat(end_date).date()
+        
+        while current_start_time < end_date_time:
```

Comment:
I'd move this logic into the internal appsflyer client so that we don't have to repeat it

Reply:
Done